clients/rospy/src/rospy/timer.py

```diff
@@ -204,7 +207,14 @@ def shutdown(self):
         Stop firing callbacks.
         """
         self._shutdown = True
-        
+        self._period = self.__period
+
+    def startup(self):
```

Comment:
The point is that "shutting down" a threading.Thread object doesn't actually shut it down, right? It just flips a bit that causes the `run` method to ignore executing its main logic block. (line 222)

The thread is technically still running until its out of scope or is assigned a NoneType value. 

My use case was having a Timer object that I could start, shutdown, retain, and restart as a condition changed.